### PR TITLE
Adds design doc for wallet extension.

### PR DESCRIPTION
**Pull request by joeldudley**

### CODE REVIEW

File: design/node_gateway.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Node gateway design
```
Comment (tudor-malene) [inline design/node_gateway.md:1]: wouldn't a better name be: "Wallet extension"?

wdyt @CaisManai ?

File: design/node_gateway.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  non-encrypting implementation of the Ethereum JSON-RPC specification
+* The node gateway is usable by any webapp or wallet type, and in particular:
+  * Hardware wallets that do not offer a decryption capability
+  * MetaMask, for which the keys are only available when running in the browser
+
+## Design
+
+The node gateway is a local server application that serves two endpoints:
+
+* An endpoint for registering new *viewing keys*
```
Comment (tudor-malene) [inline design/node_gateway.md:25]: for "managing" viewing keys.
They have to be:
- created
- signed by the account owner key
- eventually: revoked,  see expiration, etc

File: design/node_gateway.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+* An endpoint that meets the Ethereum JSON-RPC specification
+
+The node gateway also maintains an RPC connection to the Obscuro host.
```
Comment (tudor-malene) [inline design/node_gateway.md:28]: to one or multiple Obscuro nodes

File: design/node_gateway.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+### Viewing-keys endpoint
+
+This endpoint serves a webpage where the end user can generate new viewing keys for their account. A fresh keypair is 
+generated, the private key is stored locally by the node gateway, and the public key is sent to the enclave via the
```
Comment (tudor-malene) [inline design/node_gateway.md:33]: the private key is stored locally by the node gateway,
then Metamask ( or equivalent wallet), is asked to sign a payload containing this public key and some metadata, and then it is sent to the obscuro node.

**Review by tudor-malene — CHANGES_REQUESTED** [review]
looks good. Left a few small comments about the signing step

File: design/node_gateway.md (hunk 1 above)
Comment (joeldudley) [reply design/node_gateway.md]: Changed.

**Review by tudor-malene — APPROVED** [review]
Based on the discussion, this can be merged I'd say

**Pull request merged**